app/serializers/user_bookmark_base_serializer.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require_relative 'post_item_excerpt'
+
+class UserBookmarkBaseSerializer < ApplicationSerializer
+  include PostItemExcerpt
+
+  attributes :id,
+             :created_at,
+             :updated_at,
+             :name,
+             :reminder_at,
+             :pinned,
+             :title,
+             :fancy_title,
+             :bookmarkable_id,
+             :bookmarkable_type,
+             :bookmarkable_user_username,
+             :bookmarkable_user_avatar_template,
+             :bookmarkable_user_name,
+             :bookmarkable_url
+
+  def title
+    object.name
+  end
+
+  def fancy_title
+    object.name
+  end
+
+  def cooked
+    "-"
+  end
```

Comment:
It doesn't seem like this method is being used.

Reply:
This is a base class, this method should be overridden. Maybe I should throw `NotImplementedError` for these.

Reply:
O is this meant to override the method in `PostItemExcerpt`?

Reply:
No...basically each of the `UserXXXBookmarkSerializer` serializers are supposed to override it with their own thing. For example in chat the serializer looks like this:

```
class UserChatMessageBookmarkSerializer < UserBookmarkBaseSerializer
  attr_reader :chat_message

  def initialize(obj, chat_message, opts)
    # what is scope??? baby don't hurt me
    super(obj, opts)
    @chat_message = chat_message
  end

  def title
    fancy_title
  end

  def fancy_title
    @fancy_title ||= chat_message.chat_channel.name
  end

  def cooked
    chat_message.cooked
  end

  def bookmarkable_user
    @bookmarkable_user ||= chat_message.user
  end

  def bookmarkable_url
    chat_message.url
  end
end
```

(Note I don't have anything pushed for chat or other plugins just yet)

Reply:
> basically each of the UserXXXBookmarkSerializer serializers are supposed to override it with their own thing.

I see so this is an abstract factory design pattern. In this case, I think the abstract class should just specify what methods each child needs to implement and then raise the appropriate error if the methods are not implemented.

Reply:
Okay cool makes sense, will add the `NotImplementedError`s :+1: